include/gc/Dialect/CPURuntime/IR/CPURuntimeOps.td

```diff
@@ -29,5 +28,57 @@ def CPURuntime_PrintfOp : CPURuntime_Op<"printf", [MemoryEffects<[MemWrite]>]>,
   }];
 }
 
+def CPURuntime_AllocOp : CPURuntime_Op<"alloc", [
+    AttrSizedOperandSegments,
+    MemoryEffects<[MemAlloc<DefaultResource>]>
+  ]> {
+
+  let summary = "CPURuntime memory allocation operation in a stack way.";
+  let description = [{
+    The `cpuruntime.alloc` operation allocates a region of memory on the CPU. It is
+    similar to the `std.alloc` op.
```

Comment:
memref.alloc? Please also note that the alloc/free must follow the FILO order as a stack. It is not that similar with normal memref.alloc. We also need to tell developers that thread-local is a performance hint, used in parallel-for environment,